metaflow/client/core.py

```diff
@@ -877,7 +877,7 @@ def successful(self):
             True if the task completed successfully and False otherwise
         """
         try:
-            return self['_success'].data
```

Comment:
how does this work if `_success` or `_task_ok` is False? This PR seems to change the semantics